meteor/client/ui/DndListItemWrapper.tsx

```diff
@@ -0,0 +1,87 @@
+import { DragDropItemTypes } from './DragDropItemTypes'
+import React, { useRef } from 'react'
+import { DragSourceMonitor, DropTargetMonitor, useDrag, useDrop, XYCoord } from 'react-dnd'
+import { Identifier } from 'dnd-core'
+import { faGripLines } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+interface IDndItemWrapperProps {
+	index: number
+	dndType: DragDropItemTypes
+	listItem: any
+	className: string
+
+	onDropOutside: () => void
+	moveHandler: (dragIndex: number, hoverIndex: number) => void
+}
+
+interface DraggableItem {
+	index: number
+	type: DragDropItemTypes
+}
+
+export const DndListItemWrapper: React.FunctionComponent<IDndItemWrapperProps> = (props: IDndItemWrapperProps) => {
+	const ref = useRef<HTMLTableRowElement>(null)
+	const index = props.index
+
+	const [{ handlerId }, drop] = useDrop<DraggableItem, void, { handlerId: Identifier | null }>({
+		accept: props.dndType,
+		collect: (monitor: DropTargetMonitor) => ({ handlerId: monitor.getHandlerId() }),
+		hover(variant: DraggableItem, monitor: DropTargetMonitor) {
```

Comment:
Is `variant` referring to a ShowStyleVariant here? If yes, it shouldn't because the wrapper don't know what item it is wrapping

Reply:
It is just a generic item for dragging, will be changed!

Reply:
Changed!